pkg/codegen/properties.go

```diff
@@ -194,6 +194,27 @@ func getType(name string, schema *schemas.JSONSchema, required *bool, typePrefix
 			for k := range schema.Properties {
 				typeStr = strcase.ToCamel(k)
 			}
+		case schema.AdditionalProperties.JSONSchema != nil:
+			if len(schema.Properties) > 0 {
+				return "", fmt.Errorf("cannot determine the type of object %s: properties and additionalProperties are both defined", name)
+			}
+			if schema.AdditionalProperties.JSONSchema.Properties != nil {
+				return "", fmt.Errorf("cannot determine the type of object %s: a sub-object is defined in 'additionalProperties', which is currently not supported, please report this issue in https://github.com/srdtrk/go-codegen", name)
```

Comment:
```suggestion
			if schema.AdditionalProperties.JSONSchema.Ref == nil {
				return "", fmt.Errorf("cannot determine the type of object %s: a non-ref schema is defined in 'additionalProperties', which is currently not supported, please report this issue in https://github.com/srdtrk/go-codegen", name)
```

I think we may want to change the assertion to be more strict like this. What do you think?

Reply:
We can be less restrictive than that because `additionalProperties` could be a primitive type : `"additionalProperties": { "type": "string" }` and not only a ref, I'm sorry I didn't mention that on my last comment (https://github.com/srdtrk/go-codegen/pull/98#discussion_r1656657061).

| **JSON Schema**                                                                          | **Compatibility** | **Result**                  |
|:----------------------------------------------------------------------------------------:|:-----------------:|:---------------------------:|
| `"additionalProperties": false`                                                          | ✅                 | -                           |
| `"additionalProperties": true`                                                           | ✅                 | `map[string]any`            |
| `"additionalProperties": { "type": "string"}`                                            | ✅                 | `map[string]string`         |
| `"additionalProperties": {"$ref": "#/definitions/Value"}`                                | ✅                 | `map[string]Value`          |
| `"additionalProperties": {"type": "object", "properties": { "foo": {"type": "string"}}}` | ❌                 | Error                       |
| `"additionalProperties": {"type": "object", "additionalProperties": true}`               | ✅                 | `map[string]map[string]any` |


 It's working nice on the given test data `map-test.json`. I choose checking properties since if it's an object, it should contains properties.

Reply:
Makes sense